recursive_art.py

```diff
@@ -20,8 +23,26 @@ def build_random_function(min_depth, max_depth):
         (See the assignment writ-eup for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
+    list = []
+    functionsnoxy = ["avg","prod","cos_pi","sin_pi","square","cube"]
+    functionsxy = ["x","y"]
+    functionsall = functionsxy + functionsnoxy
+    randomfunction = random.choice(functionsnoxy)
+    if min_depth <= 1:
+        randomfunction = random.choice(functionsall)
+        if max_depth == 1:
+            randomfunction = random.choice(functionsxy)
+    if randomfunction in ["avg","prod"]:
+        list.append(randomfunction)
+        list.append(build_random_function(min_depth-1,max_depth-1))
+        list.append(build_random_function(min_depth-1,max_depth-1))
+    if randomfunction in ["cos_pi","sin_pi","square","cube"]:
+        list.append(randomfunction)
+        list.append(build_random_function(min_depth-1,max_depth-1))
+    if randomfunction in ["x","y"]:
+        list.append(randomfunction)
+    return list
+
```

Comment:
For the two if statements from the bottom, I think you can make it as `elif` statement. That would make the code to jump right to the return if the `randomfunction` was 'avg' or 'prod'. (Just minor details)
Also, I think `list` is a reserved keyword for python language, so I would recommend not to use list as variable names. Sometimes you might see unexpected behaviors